src/gui/AutomationPatternView.cpp

```diff
@@ -270,17 +269,22 @@ void AutomationPatternView::paintEvent( QPaintEvent * )
 	if( !( m_pat->getTrack()->isMuted() || m_pat->isMuted() ) )
 		c = styleColor;
 	else
-		c = QColor( 80, 80, 80 );
+		c = mutedColor();
 
 	if( isSelected() == true )
 	{
-		c.setRgb( qMax( c.red() - 128, 0 ), qMax( c.green() - 128, 0 ), 255 );
+		c = selectedColor();
 	}
 
 	lingrad.setColorAt( 1, c.darker( 300 ) );
 	lingrad.setColorAt( 0, c );
-
-	p.setBrush( lingrad );
+	
+	if ( gradient() ) {
+		p.setBrush( lingrad );
+	} else {
+		p.setBrush( c );
+	}
+	
```

Comment:
Again, of no consequence to you, but this logic could really be cleaned up... Food for thought. :beers: 

```diff
- if( m_needsUpdate == false )
-	{
-		QPainter p( this );
-		p.drawPixmap( 0, 0, m_paintPixmap );
-		return;
-	}

+ QPainter painter(this);
+
+ if (!m_needsUpdate)
+ {
+    painter.drawPixmap(0, 0, m_paintPixmap);
+    return;
+ }

-	QPainter _p( this );
-	const QColor styleColor = _p.pen().brush().color();
-
	m_needsUpdate = false;

-	if( m_paintPixmap.isNull() == true || m_paintPixmap.size() != size() )
-	{
-		m_paintPixmap = QPixmap( size() );
-	}

+	// init if null/empty
+	m_paintPixmap = m_paintPixmap.isNull() == true || m_paintPixmap.size() != size() ? QPixmap(size()) : m_paintPixmap;

	QPainter p( &m_paintPixmap );

	QLinearGradient lingrad( 0, 0, 0, height() );
	QColor c;

-	if( !( m_pat->getTrack()->isMuted() || m_pat->isMuted() ) )
-		c = styleColor;
-	else
-		c = mutedColor();
-
-	if( isSelected() == true )
-	{
-		c = selectedColor();
-	}

+	// priority:  selected, muted, normal
+	bool muted = m_pat->getTrack()->isMuted() || m_pat->isMuted();
+	c = isSelected() ? selectedColor() : (muted ? mutedColor() : painter.pen().brush().color());

	lingrad.setColorAt( 1, c.darker( 300 ) );
	lingrad.setColorAt( 0, c );
	
-	if ( gradient() ) {
-		p.setBrush( lingrad );
-	} else {
-		p.setBrush( c );
-	}

+	p.setBrush(gradient() ? lingrad : c);
```